draco/analysis/delay.py

```diff
@@ -846,34 +889,126 @@ def fourier_matrix_c2r(N, fsel=None):
+    """
+
+    return array.astype(np.complex128, order="C").view(np.float64).copy()
+
+
+def _alternating_real_to_complex(array):
+    """View real numbers as complex, interpreted as alternating real and imag. components.
+
+    Parameters
+    ----------
+    array : array_like
+        Input array of real numbers. Last axis must have even number of elements.
+
+    Returns
+    -------
+    out : array_like
+        Output array of complex numbers, derived from compressing the last axis (if
+        `array` has `N` real elements in the last axis, `out` will have `N/2` complex
+        elements).
+    """
+
+    return (array[..., 0::2] + 1.0j * array[..., 1::2]).astype(np.complex128)
+
+
 def delay_spectrum_gibbs(
-    data, N, Ni, initial_S, window=True, fsel=None, niter=20, rng=None
+    data,
+    N,
+    Ni,
+    initial_S,
+    window="nuttall",
+    fsel=None,
+    niter=20,
+    rng=None,
+    complex_input=False,
 ):
-    """Estimate the delay spectrum by Gibbs sampling.
+    """Estimate the delay power spectrum by Gibbs sampling.
 
     This routine estimates the spectrum at the `N` delay samples conjugate to
-    the frequency spectrum of ``N/2 + 1`` channels. A subset of these channels
-    can be specified using the `fsel` argument.
+    an input frequency spectrum with ``N/2 + 1`` channels (if the delay spectrum is
+    assumed real) or `N` channels (if the delay spectrum is assumed complex).
+    A subset of these channels can be specified using the `fsel` argument.
 
     Parameters
     ----------
     data : np.ndarray[:, freq]
         Data to estimate the delay spectrum of.
     N : int
-        The length of the output delay spectrum. There are assumed to `N/2 + 1`
-        total frequency channels.
+        The length of the output delay spectrum. There are assumed to be `N/2 + 1`
+        total frequency channels if assuming a real delay spectrum, or `N` channels
+        for a complex delay spectrum.
     Ni : np.ndarray[freq]
         Inverse noise variance.
     initial_S : np.ndarray[delay]
-        The initial delay spectrum guess.
-    window : bool, optional
-        Apply a Nuttall apodisation function. Default is True.
+        The initial delay power spectrum guess.
+    window : one of {'nuttall', 'blackman_nuttall', 'blackman_harris', None}, optional
+        Apply an apodisation function. Default: 'nuttall'.
     fsel : np.ndarray[freq], optional
         Indices of channels that we have data at. By default assume all channels.
     niter : int, optional
         Number of Gibbs samples to generate.
     rng : np.random.Generator, optional
         A generator to use to produce the random samples.
+    complex_input : bool, optional
+        If True, assume input data arose from a complex timestream. If False, assume
+        input data arose from a real timestream, such that the first and last frequency
+        channels have purely real values. Default: False.
```

Comment:
I find the name of this parameter a bit confusing, as I keep thinking it refers to the task input, but it really refers to some abstract input that we never see. There's probably a more descriptive name. `complex_timedomain`?